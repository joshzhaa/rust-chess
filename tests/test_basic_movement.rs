// for some reason, rust doesnt support integration tests that just print output and compare to see
// if it is correct. this seems like it would be a very common use case, and I don't understand why
// cargo doesn't seem to support it. although its always possible i havent read enough of the book
mod game;
mod board;
mod piece;

fn main() {
    let mut game = game::Game::new();
    game.draw();
    let pos = board::Vector(5, 7);
    println!("{}", pos.to_notation().unwrap());
    game.move_piece_unchecked(&pos, &board::Vector(0, 0));
    game.draw();
    game.move_piece_unchecked(&board::Vector(1, 0), &board::Vector(1, 7));
    game.draw();
    game.move_piece_unchecked(&board::Vector(4, 0), &board::Vector(3, 0));
    game.draw();
    game.move_piece_unchecked(&board::Vector(4, 7), &board::Vector(3, 7));
    game.draw();
    println!("SELECTING");
    game.select(board::Vector(3, 0));
    game.draw();
    game.move_piece_unchecked(&board::Vector(3, 0), &board::Vector(3, 1));
    game.draw();
    println!("REWINDING");
    game.rewind(1);
    game.draw();
}
