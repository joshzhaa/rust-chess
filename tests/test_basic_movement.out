+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:2 | B:2 | Q:2 | K:2 | B:2 | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:1 | N:1 | B:1 | Q:1 | K:1 | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move:  :0 to a1
player to move: 1
halfmove counter: 1
f8
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:2 | B:2 | Q:2 | K:2 |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| B:2 | N:1 | B:1 | Q:1 | K:1 | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: B:2 to a1
player to move: 2
halfmove counter: 2
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:1 | B:2 | Q:2 | K:2 |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| B:2 |     | B:1 | Q:1 | K:1 | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: N:1 to b8
player to move: 1
halfmove counter: 3
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:1 | B:2 | Q:2 | K:2 |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| B:2 |     | B:1 | K:1 |     | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: K:1 to d1
player to move: 2
halfmove counter: 4
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:1 | B:2 | K:2 |     |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| B:2 |     | B:1 | K:1 |     | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: K:2 to d8
player to move: 1
halfmove counter: 5
SELECTING
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:1 | B:2 | K:2 |     |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     | * * |     |     |     |
| B:2 |     | B:1 | K:1 |     | B:1 | N:1 | R:1 |
|     |     |     |     | * * |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: K:2 to d8
player to move: 1
halfmove counter: 5
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:1 | B:2 | K:2 |     |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | K:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| B:2 |     | B:1 |     |     | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: K:1 to d2
player to move: 2
halfmove counter: 6
REWINDING
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| R:2 | N:1 | B:2 | K:2 |     |     | N:2 | R:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 | P:2 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 | P:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
|     |     |     |     |     |     |     |     |
| B:2 |     | B:1 | K:1 |     | B:1 | N:1 | R:1 |
|     |     |     |     |     |     |     |     |
+-----+-----+-----+-----+-----+-----+-----+-----+
most recent move: K:2 to d8
player to move: 1
halfmove counter: 5
